pylint/pyreverse/writer.py

```diff
@@ -44,94 +52,104 @@ def write(self, diadefs):
             self.printer.emit_edge(
-                rel.from_object.fig_id, rel.to_object.fig_id, **self.imp_edges
+                rel.from_object.fig_id,
+                rel.to_object.fig_id,
+                type_=EdgeType.IMPLEMENTS,
             )
         # generate associations
         for rel in diagram.get_relationships("association"):
             self.printer.emit_edge(
                 rel.from_object.fig_id,
                 rel.to_object.fig_id,
                 label=rel.name,
-                **self.association_edges,
+                type_=EdgeType.ASSOCIATION,
             )
 
-    def set_printer(self, file_name, basename):
+    def set_printer(self, file_name: str, basename: str) -> None:
         """set printer"""
         raise NotImplementedError
 
-    def get_title(self, obj):
+    def get_title(self, obj: DiagramEntity) -> str:
         """get project title"""
         raise NotImplementedError
 
-    def get_values(self, obj):
-        """get label and shape for classes."""
+    def get_package_properties(self, obj: PackageEntity) -> NodeProperties:
+        """get label and shape for packages."""
         raise NotImplementedError
 
-    def close_graph(self):
-        """finalize the graph"""
+    def get_class_properties(self, obj: ClassEntity) -> NodeProperties:
+        """get label and shape for classes."""
         raise NotImplementedError
 
+    def save(self) -> None:
+        """write to disk"""
+        self.printer.generate(self.file_name)
+
 
 class DotWriter(DiagramWriter):
     """write dot graphs from a diagram definition and a project"""
 
-    def __init__(self, config):
-        styles = [
-            dict(arrowtail="none", arrowhead="open"),
-            dict(arrowtail="none", arrowhead="empty"),
-            dict(arrowtail="node", arrowhead="empty", style="dashed"),
-            dict(
-                fontcolor="green", arrowtail="none", arrowhead="diamond", style="solid"
-            ),
-        ]
-        DiagramWriter.__init__(self, config, styles)
-
-    def set_printer(self, file_name, basename):
+    def set_printer(self, file_name: str, basename: str) -> None:
         """initialize DotWriter and add options for layout."""
-        layout = dict(rankdir="BT")
-        self.printer = DotBackend(basename, additional_param=layout)
+        self.printer = DotPrinter(basename, layout=Layout.BOTTOM_TO_TOP)
         self.file_name = file_name
 
-    def get_title(self, obj):
+    def get_title(self, obj: DiagramEntity) -> str:
         """get project title"""
         return obj.title
 
-    def get_values(self, obj):
+    def get_package_properties(self, obj: PackageEntity) -> NodeProperties:
+        """get label and shape for packages."""
+        return NodeProperties(
+            label=self.get_title(obj),
+            color="black",
+        )
+
+    def get_class_properties(self, obj: ClassEntity) -> NodeProperties:
         """get label and shape for classes.
```

Comment:
``get_values`` is now ``get_class_properties``.

``get_package_properties`` is currently trivial and the color is hardcoded, but this will change when the new feature for automatic coloring are merged.